src/isar/state_machine/state_machine.py

```diff
@@ -266,31 +273,67 @@ def stop_mission(self):
         self.queues.stop_mission.output.put(deepcopy(message))
         self.logger.info(message)
         if not failure:
+            self.current_mission.status = MissionStatus.Cancelled
             self.mission_in_progress = False
+            for task in self.current_mission.tasks:
+                for step in task.steps:
+                    if step.status == StepStatus.NotStarted:
+                        step.status = StepStatus.Cancelled
+                if task.status == TaskStatus.NotStarted:
+                    task.status = TaskStatus.Cancelled
```

Comment:
Is the status changed to cancelled only if it's not started?
If a task is in progress when the mission is cancelled, should it not also be marked as cancelled?

Reply:
Good catch